lib/lib-upload/src/main/java/org/sagebionetworks/upload/multipart/CloudServiceMultipartUploadDAO.java

```diff
@@ -1,57 +1,46 @@
-package org.sagebionetworks.upload.multipart;
-
-import java.net.URL;
-
-import org.sagebionetworks.repo.model.file.AddPartRequest;
-import org.sagebionetworks.repo.model.file.CompleteMultipartRequest;
-import org.sagebionetworks.repo.model.file.MultipartUploadRequest;
-
-/**
- * Abstraction for S3 multi-part upload.
- *
- */
-public interface S3MultipartUploadDAO {
-
-	/**
-	 * Start a multi-part upload.
-	 * @param bucket
-	 * @param key
-	 * @param request
-	 * @return
-	 */
-	public String initiateMultipartUpload(String bucket, String key, MultipartUploadRequest request);
-
-	/**
-	 * Create a pre-signed URL to A pre-signed URL to upload a part of multi-part file upload.
-	 * @param bucket
-	 * @param partKey
-	 * @param contentType Optional parameter.  Sets the expected content-type of the request. The content-type is included in
-     * the signature.
-	 * @return
-	 */
-	public URL createPreSignedPutUrl(String bucket, String partKey, String contentType);
-
-	/**
-	 * Add a part to a multi-part upload.s
-	 * @param bucket
-	 * @param key
-	 * @param partKey
-	 * @param partMD5Hex
-	 */
-	public void addPart(AddPartRequest request);
-
-	/**
-	 * Delete an object for the given bucket and key.
-	 * @param bucket
-	 * @param key
-	 */
-	public void deleteObject(String bucket, String key);

```

Comment:
In all cases, delete object can occur atomically when addPart succeeds. The method does not need to be exposed (and hides a distinction between how the AWS and Google upload implementations differ)